src/Accounts/Authentication/Sanitizer/Services/DefaultSanitizerService.cs

```diff
@@ -44,67 +45,16 @@ internal class DefaultSanitizerService : ISanitizerService
             { "Microsoft.Azure.Storage.File.CloudFileDirectory", new[] { "Parent" } },
         };
 
-        private static readonly IEnumerable<string> SensitiveDataPatterns = new List<string>()
-        {
-            // AAD client app, most recent two versions.
-            @"\b" // pre-match
-          + @"[0-9A-Za-z-_~.]{3}7Q~[0-9A-Za-z-_~.]{31}\b|\b[0-9A-Za-z-_~.]{3}8Q~[0-9A-Za-z-_~.]{34}" // match
-          + @"\b", // post-match
-            
-            // Prominent Azure provider 512-bit symmetric keys.
-            @"\b" // pre-match
-          + @"[0-9A-Za-z+/]{76}(APIM|ACDb|\+(ABa|AMC|ASt))[0-9A-Za-z+/]{5}[AQgw]==" // match
-          + @"", // post-match
-
-            // Prominent Azure provider 256-bit symmetric keys.
-            @"\b" // pre-match
-          + @"[0-9A-Za-z+/]{33}(AIoT|\+(ASb|AEh|ARm))[A-P][0-9A-Za-z+/]{5}=" // match
-          + @"", // post-match
-            
-            // Azure Function key.
-            @"\b" // pre-match
-          + @"[0-9A-Za-z_\-]{44}AzFu[0-9A-Za-z\-_]{5}[AQgw]==" // match
-          + @"", // post-match
-
-            // Azure Search keys.
-            @"\b" // pre-match
-          + @"[0-9A-Za-z]{42}AzSe[A-D][0-9A-Za-z]{5}" // match
-          + @"\b", // post-match
-            
-            // Azure Container Registry keys.
-            @"\b" // pre-match
-          + @"[0-9A-Za-z+/]{42}\+ACR[A-D][0-9A-Za-z+/]{5}" // match
-          + @"\b", // post-match
-            
-            // Azure Cache for Redis keys.
-            @"\b" // pre-match
-          + @"[0-9A-Za-z]{33}AzCa[A-P][0-9A-Za-z]{5}=" // match
-          + @"", // post-match
-            
-            // NuGet API keys.
-            @"\b" // pre-match
-          + @"oy2[a-p][0-9a-z]{15}[aq][0-9a-z]{11}[eu][bdfhjlnprtvxz357][a-p][0-9a-z]{11}[aeimquy4]" // match
-          + @"\b", // post-match
-            
-            // NPM author keys.
-            @"\b" // pre-match
-          + @"npm_[0-9A-Za-z]{36}" // match
-          + @"\b", // post-match
-        };
+        private readonly SecretMasker _secretMaster = new SecretMasker(WellKnownRegexPatterns.HighConfidenceMicrosoftSecurityModels, generateCorrelatingIds: true);
 
         public bool TrySanitizeData(string data, out string sanitizedData)
         {
             sanitizedData = string.Empty;
 
             if (!string.IsNullOrWhiteSpace(data))
             {
-                foreach (var pattern in SensitiveDataPatterns)
-                {
-                    if (Regex.IsMatch(data, pattern))
-                    {
-                        return true;
-                    }
-                }
+                var detections = _secretMaster.DetectSecrets(data);
```

Comment:
>DetectSecrets [](http://example.com/codeflow?start=47&length=13)

so what you are doing here is only running the detections logic. and then reporting if you found at least one problem.

is that what you want? 

the `MaskSecrets(string) : string` API will accept an input and return a string with all detected secrets redacted from it, if that's what you need.

Reply:
> > DetectSecrets
> 
> so what you are doing here is only running the detections logic. and then reporting if you found at least one problem.
> 
> is that what you want?
> 
> the `MaskSecrets(string) : string` API will accept an input and return a string with all detected secrets redacted from it, if that's what you need.

Yes, I just need the detection logic instead of masking them.

Reply:
ok, so you return a bool that basically tells the caller 'there is one or more secrets in the string that you provided me. I note that you aren't doing anything useful with the `sanitizedData` argument.

do you plan to do something with it in the future? the name of this method strongly suggests that sanitization should have occurred. :)

looking at the callers to this method, I see that you produce telemetry on detecting a secret in various data. that's good! your implementation would be improved if you flowed more detection data back to the caller. specifically, the kind of secret and its correlating id would be helpful. maybe for a future change?

Reply:
Yes, here the redaction is a placeholder for future development. I will log the secrets pattern in the telemetry. What data should I log? The correlation id and/or pattern id (SECXXX/XXX) or something else?